src/index.ts

```diff
@@ -21,19 +21,17 @@ export interface InstructionMap {
   [key: string]: (element: Element) => void;
 }
 
-/**
- * A collection of tagName to errors.
- */
-type ErrorMultimap = Map<string, string[]>;
-
-export type Result<T> =
-  | {type: 'success'; value: T}
-  | {type: 'failure'; error: ErrorMultimap};
+function defaultHandleError(tagName, e: Error) {
+  const errorMsg =
+    `[bento-compiler]: Error thrown while rendering <${tagName}>\n` + e.stack;
+  throw new Error(errorMsg);
+}
 
 export function renderAst(
   tree: TreeProto,
-  instructions: InstructionMap
-): Result<TreeProto> {
+  instructions: InstructionMap,
+  {handleError = defaultHandleError} = {}
```

Comment:
Why have this be an object instead of just `handleError`?

Reply:
Its a pretty common pattern to have the last object of a function be an object bag. For now, we only have a single option, but I could easily imagine this growing.  🤷

Reply:
I'd lean away from an object bag until it holds more than one thing, but I don't feel too strongly

Reply:
> I'd lean away from an object bag until it holds more than one thing, but I don't feel too strongly

Going to leave as-is b/c of inertia and 0 consequence either way